src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/samplers/MetricsDBFileSampler.java

```diff
@@ -0,0 +1,83 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.samplers;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.AppContext;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metricsdb.MetricsDB;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.metrics.ReaderMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.stats.collectors.SampleAggregator;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.stats.emitters.ISampler;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.util.Objects;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class MetricsDBFileSampler implements ISampler {
+  private static final Logger LOG = LogManager.getLogger(MetricsDBFileSampler.class);
+  private final AppContext appContext;
+
+  public MetricsDBFileSampler(final AppContext appContext) {
+    Objects.requireNonNull(appContext);
+    this.appContext = appContext;
+  }
+
+  @Override
+  public void sample(SampleAggregator sampleCollector) {
+    String metricsdbPrefix = MetricsDB.getFilePrefix();
+
+    Path targetDirectoryPath = Paths.get(metricsdbPrefix).getParent();
+
+    String metricsdbPattern = "regex:" + metricsdbPrefix + "\\d+";
+    String metricsdbTarPattern = "regex:" + metricsdbPrefix.substring(0, metricsdbPrefix.length() - 1)
+        + ".tar.\\d+-\\d+-\\d+-\\d+-\\d+.gz";
+    PathMatcher metricsdbMatcher = FileSystems.getDefault().getPathMatcher(metricsdbPattern);
+    PathMatcher metricsdbTarMatcher = FileSystems.getDefault().getPathMatcher(metricsdbTarPattern);
+    DirectoryStream.Filter<Path> metricsdbFilter = metricsdbMatcher::matches;
+    DirectoryStream.Filter<Path> metricsdbTarFilter = metricsdbTarMatcher::matches;
+
+    int numUncompressedMetricsdbFiles = 0;
+    long sizeUncompressedMetricsdbFiles = 0;
+    int numMetricsdbFiles = 0;
+    long sizeMetricsdbFiles = 0;
+
+    try (DirectoryStream<Path> stream = Files.newDirectoryStream(targetDirectoryPath, metricsdbFilter)) {
+      for (Path entry : stream) {
+        try {
+          sizeUncompressedMetricsdbFiles += Files.size(entry);
+          numUncompressedMetricsdbFiles += 1;
+        } catch (IOException e) {
+          LOG.error("Failed to access metricsdb file - {}", entry);
+        }
+      }
+    } catch (IOException e) {
+      LOG.error("Failed to access metricsdb directory - {}", targetDirectoryPath);
+    }
```

Comment:
Can we make this logic into a util method that both metricsdbTarFilter and metricsdbFilter can use?

Reply:
I don't think that'll work well since we want it the logic to produce two variables (sizeUncompressedMetricsdbFiles and numUncompressedMetricsdbFiles). Also, the duplication here doesn't seem like too much to require factoring out the logic.